README.md

```diff
@@ -120,17 +124,39 @@ jobs:
             - tag: ./node_modules/.bin/template-tag --name templateName --version 1.2.3 --tag stable
 ```
 
-Create a Screwdriver pipeline with your template repo and start the build to validate and publish it.
-
-To update a Screwdriver template, make changes in your SCM repository and rerun the pipeline build.
-
-`template-validate` can print a result as json by passing `--json` option to the command.
+`template-tag` can print a result as json by passing `--json` option to the command.
 
 ```
-$ ./node_modules/.bin/template-publish --json --name templateName --version 1.2.3 --tag stable
+$ ./node_modules/.bin/template-tag --json --name templateName --version 1.2.3 --tag stable
 {"name":"templateName","tag":"stable","version":"1.2.3"}
 ```
 
+##### Removing a template tag
+
+Adding the `--delete` or `-d` flag to the `template-tag` script will delete a template. This must be done in the same pipeline that published the template. You'll need to specify the template name and tag as arguments.
+
+Example `screwdriver.yaml` with validation, publishing and tagging, and tag removal as a detached job:
+
+```yaml
+shared:
+    image: node:6
+    steps:
+        - init: npm install screwdriver-template-main
+jobs:
+    main:
+        requires: [~pr, ~commit]
+        steps:
+            - validate: ./node_modules/.bin/template-validate
+    publish:
+        requires: main
+        steps:
+            - publish: ./node_modules/.bin/template-publish
+            - tag: ./node_modules/.bin/template-tag --name templateName --version 1.0.0 --tag latest
+    detached_remove_tag:
+        steps:
+            - remove: ./node_modules/.bin/template-tag --delete --name templateName --tag latest
```

Comment:
I'd prefer if the mechanism to delete a template tag was similar to how to delete a template.
Not sure if having the `--delete` flag for deleting a template or naming the script to `template-tag-remove` is better.

Reply:
I agree that there is some inconsistency. How about `tag-publish` and `tag-remove`?

Reply:
I think a lot of people are already using template tag stuff since it was already made public a while ago. It might be better/less painful for our users if we just change the removing a template mechanism

Reply:
We can just deprecate `template-tag`, but keep it as an alias for `tag-publish`. That way there's no breaking changes, but our binary names are more consistent:

```
template-remove
template-publish
tag-remove
tag-publish
```

Reply:
A lot of ppl are already using `template-tag` now. Please do not deprecate that, we are going to break ppl.

Reply:
@minz1027 We can deprecate `template-tag` without breaking people's pipelines. Suppose we split `tag.js` into `tagRemove.js` and `tagPublish.js`. We can then expose the binaries this way in `package.json`:

```json
"bin": {
  "template-publish": "./publish.js",
  "template-validate": "./validate.js",
  "template-remove": "./remove.js",
  "tag-publish": "./tagPublish.js",
  "tag-remove": "./tagRemove.js",
  "template-tag": "./tagPublish.js"
}
```

`template-tag` will still be included as a binary, so no breaking change, but it'll be deprecated.

Reply:
But going forward, if we keep both, then it's not deprecated and ppl can be confused since tag-publish and template-tag are both doing the same thing... how about `template-tag` and `template-remove-tag` for consistency?

Reply:
I've made a new commit demonstrating how we'd pull this off. Ideally we'd want some sort of feedback from a template author; @drewfish perhaps?

Reply:
After some thought, I think @minz1027's suggestion, naming the binaries `template-tag` and `template-remove-tag`, is the way to go 👍